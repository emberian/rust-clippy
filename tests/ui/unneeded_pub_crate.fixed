// run-rustfix
#![allow(dead_code)]
#![warn(clippy::unneeded_pub_crate)]

 struct Baz {
     field: u8,
}

mod outer {
    mod inner {
        // do these things _really_ need to be `pub(crate)`?
         struct Foo;
         trait Baz {
            fn the_goods(&self);
            fn secret_thing(&self);
        }
        impl Baz for crate::Baz {
            fn the_goods(&self) {}
            fn secret_thing(&self) {}
        }
         fn foo() {}
        pub(crate) fn bar() -> super::ReturnStruct {
            foo();
            let _ = Foo;
            let x = crate::Baz { field: 3 };
            x.the_goods();
            x.secret_thing();
            super::ReturnStruct {
                used_outside: 0,
                not_used_outside: 0,
            }
        }
        pub(crate) struct S1(u8, pub(crate) u8);
    }
    pub(crate) struct ReturnStruct {
        pub(crate) used_outside: u8,
         not_used_outside: u8,
    }
    pub(crate) fn main() -> (ReturnStruct, inner::S1) {
        inner::bar()
    }
}

fn main() {
    let (value_outside, s1) = crate::outer::main();
    s1.1 = 42;
    let _ = value_outside.used_outside;
}
