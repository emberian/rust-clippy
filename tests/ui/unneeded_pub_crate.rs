// run-rustfix
#![allow(dead_code)]
#![warn(clippy::unneeded_pub_crate)]

pub(crate) struct Baz {
    pub(crate) field: u8,
}

mod outer {
    mod inner {
        // do these things _really_ need to be `pub(crate)`?
        pub(crate) struct Foo;
        pub(crate) trait Baz {
            fn the_goods(&self);
            fn secret_thing(&self);
        }
        impl Baz for crate::Baz {
            fn the_goods(&self) {}
            fn secret_thing(&self) {}
        }
        pub(crate) fn foo() {}
        pub(crate) fn bar() -> super::ReturnStruct {
            foo();
            let _ = Foo;
            let x = crate::Baz { field: 3 };
            x.the_goods();
            x.secret_thing();
            super::ReturnStruct {
                used_outside: 0,
                not_used_outside: 0,
            }
        }
    }
    pub(crate) struct ReturnStruct {
        pub(crate) used_outside: u8,
        pub(crate) not_used_outside: u8,
    }
    pub(crate) fn main() -> ReturnStruct {
        inner::bar()
    }
}

fn main() {
    let value_outside = crate::outer::main();
    let _ = value_outside.used_outside;
}
