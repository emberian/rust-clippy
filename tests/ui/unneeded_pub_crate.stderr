error: pub(crate) item is never used outside of its defining module
  --> $DIR/unneeded_pub_crate.rs:6:5
   |
LL |     pub(crate) field: u8,
   |     ^^^^^^^^^^ help: consider removing pub(crate)
   |
   = note: `-D clippy::unneeded-pub-crate` implied by `-D warnings`

error: pub(crate) item is never used outside of its defining module
  --> $DIR/unneeded_pub_crate.rs:12:9
   |
LL |         pub(crate) struct Foo;
   |         ^^^^^^^^^^ help: consider removing pub(crate)

error: pub(crate) item is never used outside of its defining module
  --> $DIR/unneeded_pub_crate.rs:5:1
   |
LL | pub(crate) struct Baz {
   | ^^^^^^^^^^ help: consider removing pub(crate)

error: pub(crate) item is never used outside of its defining module
  --> $DIR/unneeded_pub_crate.rs:13:9
   |
LL |         pub(crate) trait Baz {
   |         ^^^^^^^^^^ help: consider removing pub(crate)

error: pub(crate) item is never used outside of its defining module
  --> $DIR/unneeded_pub_crate.rs:21:9
   |
LL |         pub(crate) fn foo() {}
   |         ^^^^^^^^^^ help: consider removing pub(crate)

error: pub(crate) item is never used outside of its defining module
  --> $DIR/unneeded_pub_crate.rs:36:9
   |
LL |         pub(crate) not_used_outside: u8,
   |         ^^^^^^^^^^ help: consider removing pub(crate)

error: aborting due to 6 previous errors

